packages/x-components/src/x-modules/scroll/store/module.ts

```diff
@@ -13,21 +13,17 @@ export const scrollXStoreModule: ScrollXStoreModule = {
   }),
   getters: {},
   mutations: {
-    setScrollPosition(state, { id, position }) {
-      initScrollComponentState(state.data, id);
-      state.data[id].position = position;
-    },
-    setScrollDirection(state, { id, direction }) {
-      initScrollComponentState(state.data, id);
-      state.data[id].direction = direction;
-    },
-    setScrollHasReachedEnd(state, { id, value }) {
-      initScrollComponentState(state.data, id);
-      state.data[id].hasReachedEnd = value;
-    },
-    setScrollHasReachedStart(state, { id, value }) {
-      initScrollComponentState(state.data, id);
-      state.data[id].hasReachedStart = value;
+    setScrollComponentState(state, { id, newState }) {
+      if (!state.data[id]) {
+        Vue.set<ScrollComponentState>(state.data, id, {
+          hasReachedStart: false,
```

Comment:
I think we can create a helper method called `createScrollInitialState`, or something like that, and move this to the utils folder.

Reply:
I think he did the opposite here 😅 
This was previously in a utility and he removed it and did it here. I wouldn't move it back to the utils folder as this is the only place where this is being used.

Reply:
I did exactly the opposite yeah, as it is now just one `if` and one statement, I think it is easier to keep it here. 

I had wrote a comment here in the PR explaining this, but forgot to send it 😅.